src/interface/obsidian/src/utils.ts

```diff
@@ -123,3 +123,50 @@ export async function createNoteAndCloseModal(query: string, modal: Modal, opt?:
     }
     modal.close();
 }
+
+export async function canConnectToBackend(
+    khojUrl: string,
+    khojApiKey: string,
+    showNotice: boolean = false
+): Promise<{ connectedToBackend: boolean; statusMessage: string, userEmail: string }> {
+    let connectedToBackend = false;
+    let userEmail: string = '';
+
+    if (!!khojUrl) {
+        let headers  = !!khojApiKey ? { "Authorization": `Bearer ${khojApiKey}` } : undefined;
+        await request({ url: `${khojUrl}/api/health`, method: "GET", headers: headers })
+        .then(response => {
+            connectedToBackend = true;
+            userEmail = JSON.parse(response)?.email;
+        })
+        .catch(error => {
+            connectedToBackend = false;
+            console.log(`Khoj connection error:\n\n${error}`);
+        });
+    }
+
+    let statusMessage: string = getBackendStatusMessage(connectedToBackend, userEmail, khojUrl, khojApiKey);
+    if (showNotice) new Notice(statusMessage);
+    return { connectedToBackend, statusMessage, userEmail };
+}
+
+export function getBackendStatusMessage(
+    connectedToServer: boolean,
+    userEmail: string,
+    khojUrl: string,
+    khojApiKey: string
+): string {
+    // Welcome message with default settings. Khoj cloud always expects an API key.
+    if (khojApiKey === '' && khojUrl === 'https://app.khoj.dev')
```

Comment:
You'll also want to make sure `khojApiKey` is not `null`, `undefined`, in addition to being empty string. Below should take care of it.

```suggestion
    if (khojApiKey && khojUrl === 'https://app.khoj.dev')
```

Reply:
Fixed that by forcing it to be a boolean with `!!`